test/torch/tensors/test_promise.py

```diff
@@ -0,0 +1,157 @@
+import pytest
+import torch
+import syft
+
+
```

Comment:
Is there a way we can test the asynchronous execution of these promises? Perhaps we can create a "ThreadWorker" which is like a VirtualWorker but it runs in its own thread (might have to communicate with sockets... in fact this could just be a wrapper around SocketWorker)? In theory the execution time for doing promises on a series of ThreadWorkers should be much faster?

Also it would be really amazing if we could check/test that our MPC stuff can be used to create a graph of promises. Meaning we want the ability to call .share() on a promise, then do some executions, and then call .get() (which is still a promise holding the results). Then call .keep() on the input and the whole thing should run automatically.

Reply:
I think it still needs some work before we can share Promises but I'll think about that!

And for the first part, I'm not sure I got what you mean... Do you want to compare with "classic" plans for which (I guess) the results are waited for each time the plan is called?

Reply:
No more compare with doing everything synchronously. Aka... if you .keep() a promise on VirtualWorkers, it should be slower than if you .keep() a promise on ThreadWorkers.

However, if you execute a command on a VirtualWorker or a ThreadWorker normally (without promises) they should run in the same amount of time (because your executing thread blocks and forces them to be synchronous).

Reply:
status on this?

Reply:
I think ThreadWorkers could indeed be useful for benchmarking!
But maybe it should be another PR?

Reply:
agreed